jest testing / readme update

Anzumana opened the pull request:
Added jest for testing our code
And updated the readme with a lot more usefull information.

the things that i am currently thinking about that could help.

============
would you be ok if we switch things over from gitter to slack ? have every little experience with gitter while slack i use on a daily basis.so i could make sure that i keep up with people that do have questions.

Also i was wondering what do you think about taking the project and making and organisation for it. 
I also work on the debugger for firefox. There we have an organisation and that works pretty great.
https://github.com/devtools-html/debugger.html

Once we have the organisation in place i would apply to make an open collective for this project.
As the main issue always tend to be how to find the proper time to work on those great side projects.

Anzumana committed:
jest testing / readme update

Added jest for testing our code
And updated the readme with a lot more usefull information.

sidorares commented:
Readme changes looks good ( maybe add notes about docker as an option as well )

> Added jest for testing our code

So this just calls jest, but we don't have any tests yet, right? Do you have anything in mind to put as first test?

> Also i was wondering what do you think about taking the project and making and organisation for it.

I don't mind moving it to org. Name suggestions? `node-vim-debugger` as org name? (maybe node is bad name as by moving to CDP we'll get browser client side scripts debugging from vim was well. `js-vim-debugger` ?)

Anzumana commented:
my initial thought was javascript-vim-debugger to make it easily searchable through google.
The more traction we can generate the better.
but js-vim-debugger would work for me too.
i rly like the shorthand jvd for the project to be honest :) But thats the case with both names ^^


will try to add docker notes. i am not that familiar with docker yet. that kind of makes it a little bit tricky.

for testing
the best place would probably be ensuring that the version for 7.6 is still viable with any future changes. So those would be the tests that i would like to create first. 

I also need to setup cycle ci or travis ci for the project to make sure pr can only be merged if our tests don't fail

Anzumana committed:
update readme contributor table

Anzumana commented:
hm not exactly sure why the markdown table is broken.
in my markdown preview everything seems fine.ˆˆ
will have to look into that tomorrow

sidorares commented:
what are you using for preview?

Anzumana commented:
http://marked2app.com/ by @ttscoff

Anzumana committed:
added example for testing the debugger

Anzumana committed:
sketch logo design files and png export

Anzumana committed:
update todo.txt
i am not sure that this is the best place to maintain the list of
all the things that we want to to.
github issue are somewhat good but its rly great if you have a file with
all the things that you want to do when offline

Anzumana committed:
faq md file

Anzumana committed:
update todo.txt with all contributors link

Anzumana committed:
added docs folder

Anzumana committed:
update readme

si-mikey commented on README.md:
````diff
@@ -1,3 +1,42 @@
+If you don't want to configure nvm to work with the debugger or your affraid you could break your current setup,
+just download the binary to your desktop.
+```bash
+~/Desktop/node-v7.6.0-darwin-x64/bin/node --debug-brk examples/qqq.js
````
--debug-brk and --debug are deprecated might want to refactor
```suggestion
~/Desktop/node-v7.6.0-darwin-x64/bin/node --inspect-brk examples/qqq.js
```

si-mikey commented on README.md:
````diff
@@ -1,3 +1,42 @@
+This alias works assuming you put the node file you downloaded from [npm][npm] onto your desktop
+If you want to place it somewhere else update the path accordingly
+```bash
+alias node7.6="~/Desktopnode-v7.6.0-darwin-x64/bin/node --debug-brk"
````
```suggestion
alias node7.6="~/Desktopnode-v7.6.0-darwin-x64/bin/node --inspect-brk"
```

sidorares commented:
Hey @CADBOT I'm really struggling to find time for my open source projects lately, I'd like to give a bit more attention to this but unfortunately can't promise anything

sidorares commented:
> Would you be interested in help maintaining the project?

definitely! How much you already know about structure of this project? Old v8 debugger protocol, chrome remote debugging protocol, vim-netbeans etc? Happy to answer any questions

sidorares commented:
> Is there a better way to reach you?

email is ok but if you feel discussion can benefit others it's better to do it in the open in GH issues ( if you want to discuss any private matter of security issue of course please email)
